plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotTreeUtilities.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.swtbot;
+
+import org.eclipse.swt.widgets.TreeItem;
+import org.eclipse.swtbot.swt.finder.SWTBot;
+import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException;
+import org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable;
+import org.eclipse.swtbot.swt.finder.results.Result;
+import org.eclipse.swtbot.swt.finder.results.VoidResult;
+import org.eclipse.swtbot.swt.finder.results.WidgetResult;
+import org.eclipse.swtbot.swt.finder.waits.DefaultCondition;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTree;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.eclipse.swtbot.swt.finder.widgets.TimeoutException;
+
+/**
+ * Utilities for manipulating trees.
+ */
+public class SwtBotTreeUtilities {
+
+  /**
+   * Given a tree that contains an entry with <code>itemName</code> and a direct child with a name
+   * matching <code>subchildName</code>, return its tree item.
+   *
+   * This method is useful when there is the possibility of a tree having two similarly-named
+   * top-level nodes.
+   *
+   * @param mainTree the tree
+   * @param itemName the name of a top-level node in the tree
+   * @param subchildName the name of a direct child of the top-level node (used to uniquely select
+   *        the appropriate tree item for the given top-level node name)
+   * @return the tree item corresponding to the top-level node with <code>itemName</code>that has a
+   *         direct child with <code>subchildName</code>. If there are multiple tree items that
+   *         satisfy this criteria, then the first one (in the UI) will be returned
+   *
+   * @throws IllegalStateException if no such node can be found
+   */
+  public static SWTBotTreeItem getUniqueTreeItem(final SWTBot bot, final SWTBotTree mainTree,
+      String itemName, String subchildName) {
+    for (SWTBotTreeItem item : mainTree.getAllItems()) {
+      if (itemName.equals(item.getText())) {
+        try {
+          item.expand();
+          waitUntilTreeHasText(bot, item);
+          if (item.getNode(subchildName) != null) {
+            return item;
+          }
+        } catch (WidgetNotFoundException e) {
+          // Ignore
+        }
+      }
+    }
+
+    throw new IllegalStateException(
```

Comment:
This feels wrong. Returning null might be better here, but not finding an item doesn't seem like an illegal state to me.  "Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation."